src/pages/admin/materiaalaanvragen/index.tsx

```diff
@@ -1,25 +1,275 @@
 import { Permission } from '@account/const';
+import {
+	ADMIN_MATERIAL_REQUESTS_QUERY_PARAM_CONFIG,
+	ADMIN_MATERIAL_REQUESTS_TABLE_PAGE_SIZE,
+	getAdminMaterialRequestTableColumns,
+} from '@admin/const/material-requests.const';
 import { AdminLayout } from '@admin/layouts';
 import { withAuth } from '@auth/wrappers/with-auth';
+import {
+	CP_MATERIAL_REQUEST_TYPE_FITLER_ARRAY,
+	CP_MATERIAL_REQUEST_TYPE_FITLER_RECORD,
+	CP_MATERIAL_REQUESTS_FILTER_ALL_ID,
+} from '@cp/const/material-requests.const';
+import { useGetContentPartners } from '@cp/hooks/get-content-partners';
+import {
+	Icon,
+	IconNamesLight,
+	Loading,
+	PaginationBar,
+	SearchBar,
+	sortingIcons,
+} from '@shared/components';
 import PermissionsCheck from '@shared/components/PermissionsCheck/PermissionsCheck';
+import { globalLabelKeys } from '@shared/const';
 import { getDefaultServerSideProps } from '@shared/helpers/get-default-server-side-props';
 import { renderOgTags } from '@shared/helpers/render-og-tags';
 import useTranslation from '@shared/hooks/use-translation/use-translation';
 import { DefaultSeoInfo } from '@shared/types/seo';
 
+import { useGetMaterialRequestById } from '@material-requests/hooks/get-material-request-by-id';
+import { useGetMaterialRequests } from '@material-requests/hooks/get-material-requests';
+import {
+	MaterialRequest,
+	MaterialRequestKeys,
+	MaterialRequestType,
+} from '@material-requests/types';
+
 const AdminMaterialRequests: NextPage<DefaultSeoInfo> = ({ url }) => {
-	const { tText } = useTranslation();
+	const { tHtml, tText } = useTranslation();
+	const [isTypeDropdownOpen, setIsTypeDropdownOpen] = useState(false);
+	const [isMaintainerDropdownOpen, setIsMaintainerDropdownOpen] = useState(false);
+	const [typeDropdownLabel, setTypeDropdownLabel] = useState(
+		tText('pages/admin/materiaalaanvragen/index___type-aanvraag')
+	);
+	const [maintainerDropdownLabel, setMaintainerDropdownLabel] = useState(
+		tText('pages/admin/materiaalaanvragen/index___aanbieder')
+	);
+	const [isDetailBladeOpen, setIsDetailBladeOpen] = useState(false);
+	const [currentMaterialRequest, setCurrentMaterialRequest] = useState<MaterialRequest>();
+	const [filters, setFilters] = useQueryParams(ADMIN_MATERIAL_REQUESTS_QUERY_PARAM_CONFIG);
+
+	const { data: materialRequests, isFetching } = useGetMaterialRequests({
+		isPersonal: false,
+		size: ADMIN_MATERIAL_REQUESTS_TABLE_PAGE_SIZE,
+		...(!isNil(filters.page) && { page: filters.page }),
+		...(!isNil(filters.orderProp) && { orderProp: filters.orderProp as MaterialRequestKeys }),
+		...(!isNil(filters.orderDirection) && {
+			orderDirection: filters.orderDirection as OrderDirection,
+		}),
+		search: filters.search,
+		type: filters.type as MaterialRequestType,
+		maintainerIds: filters.maintainerIds as string[],
+	});
+
+	const { data: maintainers } = useGetContentPartners();
+
+	const getMaintainers = () => {
+		const arr = [] as MenuItemInfo[];
```

Comment:
Maybe put this in a useMemo instead of calling this funciton multiple times in this file.

```
	const maintainerList = useMemo(() => {
		return [
			{
				id: 'ALL',
				label: ADMIN_MATERIAL_REQUEST_TYPE_FITLER_RECORD[ADMIN_MATERIAL_REQUESTS_FILTER_ALL_ID],
			}
			...maintainers.map(({id, label}: maintainerType): MenuItemInfo => ({id, label: name}))
		]
	}, [maintainers]);
```

This way, you can use `maintainerList` everywhere you need + it will always be up to date.

Notice that I changed a few variables since to `ADMIN_` instead of `CP_`. If you need a similar value, I'd make new once for the admin module.